src/wasm/wasm-binary.cpp

```diff
@@ -5709,13 +5709,14 @@ bool WasmBinaryBuilder::maybeVisitBrOnCast(Expression*& out, uint32_t code) {
     return false;
   }
   auto name = getBreakTarget(getU32LEB()).name;
-  auto heapType1 = getHeapType();
-  auto heapType2 = getHeapType();
+  // TODO the spec has two heaptype immediates, but the V8 prototype does not;
+  //      match that for now.
   auto* rtt = popNonVoidExpression();
-  validateHeapTypeUsingChild(rtt, heapType2);
+  if (!rtt->type.isRtt()) {
+    throwError("bad rtt for br_on_cast");
+  }
```

Comment:
Does this mean we can't parse unreachable rtts in this position? That seems bad. Would we be able to synthesize an arbitrary heap type to pass to the builder in case the rtt is unreachable instead?

Reply:
I think this is not a problem because we skip unreachable code anyhow (in `WasmBinaryBuilder::skipUnreachableCode`). That is, if the operand is unreachable, we never reach the rtt, unless I'm missing something. (If I am, the fuzzer will eventually correct us :wink: )